multi-int-store/src/main/java/com/connexta/multiintstore/common/ProductStorageManager.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.multiintstore.common;
+
+import com.connexta.multiintstore.adaptors.RetrieveResponse;
+import com.connexta.multiintstore.adaptors.StorageAdaptor;
+import com.connexta.multiintstore.common.exceptions.StorageException;
+import java.io.IOException;
+import java.net.URL;
+import java.util.UUID;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+@Slf4j
+@Service
+public class ProductStorageManager {
```

Comment:
:exclamation: I'm starting to worry we have to many layers of abstraction.
ControllerX -> StorageManagerX -> StorageAdaptorX -> RepositoryX
Is there a good case for having both StorageManagers and StorageAdaptors?

Reply:
My recommendation is to divide up the responsibility like this:
* Controllers are the interface between HTTP messages and model objects. For example, a controller might receive a multipart http request. It should convert (or rely on registered converters) to create a model object like Product or Metadata. 
* Controllers get models or store models by passing them to the StorageManager. The storage manager is composed of multiple StorageAdaptors, each responsible for communicating with a different persistence mechanism. There is only one StorageManager class and it is concrete. There is only one instance of StorageManager in the application.
* StorageAdaptors get models or store models by working with a Spring Data repository (like Solr), or working with a client like the S3Client. StorageAdaptor is an interface and there is a concrete subclass for each type of storage.

Reply:
I think we should address this in another PR. Feel free to create an issue.